### PR TITLE
Pdf view new

### PR DESCRIPTION
**Pull request by ngprnk**

### CODE REVIEW

File: src/views/components/map/Map.web.js [hunk 1]
```diff
@@ -1,16 +1,510 @@
...
+      height: string, 
+      width: string,
+
+    }
+
+    state = {
+      error: null,
+    };
+
+    UNSAFE_componentWillMount() {
```
Comment (zcallan) [inline src/views/components/map/Map.web.js]: We shouldn't use deprecated methods - can we please update this to either be in cdM or outside the class?

File: src/views/components/map/Map.web.js [hunk 2]
```diff
@@ -1,16 +1,510 @@
...
+      this.setupGoogleApi();
+    }
+    
+    componentWillUnmount() {
+      if ( this.map ) {
+        this.map = null;
+      }
+    }
+
+    setupGoogleApi = () => {
```
Comment (zcallan) [inline src/views/components/map/Map.web.js]: We already do this in GoogleProvider - are we able to reuse that function / remove this as it shouldn't be needed? I want to try eliminate duplicated code.

File: src/views/components/map/Map.web.js [hunk 3]
```diff
@@ -1,16 +1,510 @@
...
+          `https://maps.googleapis.com/maps/api/js?key=${config.google.apiKey}`;
+  
+        script.async = true;
+        script.id = 'google-api';
+        document.body.appendChild( script );
+      }
+    };
+
+    setup = () => {
+      if ( typeof google === 'object' ) {
```
Comment (zcallan) [inline src/views/components/map/Map.web.js]: Can we please use our internal `isObject` inside our `src/utils` directory? Simply checking if `google` is a `typeof` object still returns true if `google` is `null` or an array, so I consider it as unsafe practice.

File: src/views/components/map/Map.web.js [hunk 4]
```diff
@@ -1,16 +1,510 @@
...
+              newLng = parseFloat( checkLng );
+            }
+          }
+        }
+      }
+
+      return { lat: newLat, lng: newLng };
+    }
+
+    componentDidMount() {
```
Comment (zcallan) [inline src/views/components/map/Map.web.js]: Can we please move this function to be sorted as per the eslint config for this class? Should be under `state` but above `componentWillUnmount`.

File: src/views/components/map/Map.web.js [hunk 5]
```diff
@@ -1,16 +1,510 @@
...
+                }
+              }
+            });
+          }
+          else {
+            this.setAlert( 'Address not found' );
+          }
+        });
+      }
+      else if ( typeof address === 'object' && address.lat && address.lng ) {
```
Comment (zcallan) [inline src/views/components/map/Map.web.js]: Also here can we use `isObject` from our `utils`?

File: src/views/components/map/Map.web.js [hunk 6]
```diff
@@ -1,16 +1,510 @@
...
+      this.setup();
+      this.drawPolygons();
+      this.drawPolylines();
+    }
+
+    checkAddressFormat = ( geocoder, value, callback ) => {
+      let address = value;
+
+      if ( typeof address === 'string' || 
+      ( typeof address === 'object' &&
```
Comment (zcallan) [inline src/views/components/map/Map.web.js]: Can we please use our internal `isObject` here, and use the options called `withProperties` and pass through `['suburb', 'state]`?

File: src/views/components/map/Map.web.js [hunk 7]
```diff
@@ -1,16 +1,510 @@
...
+    componentDidMount() {
+      this.setup();
+      this.drawPolygons();
+      this.drawPolylines();
+    }
+
+    checkAddressFormat = ( geocoder, value, callback ) => {
+      let address = value;
+
+      if ( typeof address === 'string' ||
```
Comment (zcallan) [inline src/views/components/map/Map.web.js]: Can we please use our internal `isString` util here?

File: src/views/components/map/Map.web.js (hunk 1 above)
Comment (ngprnk) [reply src/views/components/map/Map.web.js]: This is a work in  progress i have been working on multiple components using the same branch. i have already made changes in my local, after this is Finalized  will push it.

**Comment by zcallan** [comment]
(ignore this comment)

**Comment by zcallan** [comment]
Closing in favour of merging in the new branch.

**Pull request closed without merging**